.github/workflows/go.yml

```diff
@@ -41,12 +41,12 @@ jobs:
       - name: Cache linter
         uses: actions/cache@v3
         with:
-          path: ~/go/bin/golangci-lint
-          key: golangci-lint-v1.51.2
+          path: ~/go/bin/staticcheck
+          key: staticcheck-v0.4.3
```

Comment:
Can you extend the commit message to say why we are doing this and what the pros and cons are for staticcheck vs golangci-lint.